### ISSUE TITLE
VST2_SDK does not contain pluginterfaces anymore

### ISSUE THREAD

**Issue by ShigekiKarita** [AuburnSounds/Dplug#305]
Hi,

As far as I know,  `vstsdk3610_11_06_2018_build_37` had the `pluginterfaces` but the latest VST2_SDK in `3611_22_10_2018_build_34` does not. Is there any solution for this problem?

**Comment by ShigekiKarita** [comment]
Maybe JUCE's header can be used instead of aeffect.h and aeffectx.h in pluginterfaces.

https://github.com/WeAreROLI/JUCE/blob/master/modules/juce_audio_processors/format_types/juce_VSTInterface.h

**Comment by p0nce** [comment]
Hello,

No one other than Steinberg is allowed to redistribute these headers, even translated. 
Dplug cannot provide these headers, nor can JUCE anymore.

There is no solution if you don't have both the headers and a VST2 Agreement with Steinberg...

**Comment by ShigekiKarita** [comment]
https://github.com/WeAreROLI/JUCE/issues/433

I confirmed that JUCE is distributing the header legally under 
>  a special agreement in place with Steinberg with regards to distributing parts of the VST2 and VST3 SDK (and keeping them in our git history)

I just wanna share this information with new VST2 developper using Dplug because we can still use JUCE's header under the JUCE license and modified dplug for JUCE's renamed definitions.

**Comment by p0nce** [comment]
Even if a legal way was possible (maybe it is in the EU), you are exposing yourself to a considerable legal risk by going against the spirit of the law. Using a translated _former_ JUCE header in D would import the JUCE dual-licence GPL + Commercial. This JUCE VST2 header only existed thanks to an arrangement that, said Jules, would not allow anyone else to redistribute these headers.

I don't know how much of Dplug is GPL-compatible, I assume it could be. However your plugins would need to be GPL at the minute you touch a JUCE header, but GPL implies redistribution of source code when asked by the user, so it seems legally impossible to distribute GPL VST2 plugins (IANAL).

**tl;dr** I don't think you would be able to redistribute such a GPL VST2 plugin, it has been a known issue with JUCE + VST2 since forever (and that's why the VST3 SDK is licensed differently).

In general in Dplug, we are not willing to include JUCE copyrighted code and the policy is _to not even look_ at JUCE code in order to avoid copyright tainting, and such GPL loophole.

**Comment by ShigekiKarita** [comment]
Thanks for telling me detailed explanation. It was much harder than I expected (I only thought about the redistribution). Now I can see what you are worry about JUCE and VST SDK.  As you mentioned, their licenses and agreements are too complicated to use them freely like dplug. Thanks to your advice, I will explore new way to develop some VST3 with GPLv3 and D.

**Comment by p0nce** [comment]
Hello, VST3 is now on the roadmap #306

**Comment by abaga129** [comment]
Alternatively you could try to contact Steinberg directly and see if they would be willing to supply a version of the SDK that contains the plugininterfaces for VST2.  Maybe if you explain your situation they would understand.
